### PR TITLE
Added matomo event to the pay and submit button

### PR DESCRIPTION
**Pull request by AlisonGriffiths**
Resolves jira ticket - BI-5257
minor change

### CODE REVIEW

File: src/main/resources/templates/checkDetails.html [hunk 1]
```diff
@@ -78,9 +78,12 @@ <h2 class="govuk-heading-m" th:text="#{checkDetails.authorised.heading}"></h2>
                     </div>
                 </fieldset>
                 <div>
-                    <button data-prevent-double-click="true" id="next-button" type="submit" class="govuk-button"
+                    <button data-prevent-double-click="true" id="next-button" type="submit" class="govuk-button piwik-event"
                             name="action" value="submit"
-                            th:text="${checkDetails.paymentCharge} ? #{checkDetails.button.payAndSubmit} : #{checkDetails.button.submit}"></button>
+                            th:text="${checkDetails.paymentCharge} ? #{checkDetails.button.payAndSubmit} : #{checkDetails.button.submit}"
+                            th:attr="data-event-id=${checkDetails.paymentCharge} ? 'payment-button' : null">
```
Comment (HarryBosworth) [inline src/main/resources/templates/checkDetails.html:84]: Is the requirement to only track journeys with a payment charge?

Comment (AlisonGriffiths) [reply src/main/resources/templates/checkDetails.html]: I have modified the code to only add in the piwik-event class for payment scenarios.
We don't need to add an event for the non payment journey as we track the non payment journey via page goals

Comment (AlisonGriffiths) [reply src/main/resources/templates/checkDetails.html]: The payment journey takes us out of the service so we need to track that by adding an event to the pay and submit button.

**Review by HarryBosworth — APPROVED** [review]
Changes look okay

**Pull request merged**